app/services/concerns/xccdf/rules.rb

```diff
@@ -7,25 +7,18 @@ module Rules
 
     included do
       def save_rules
-        @rules ||= @op_rules.map do |op_rule|
-          ::Rule.from_openscap_parser(op_rule, benchmark_id: @benchmark&.id)
+        @rules ||= @op_rules.each_with_index.map do |op_rule, idx|
+          ::Rule.from_openscap_parser(op_rule, precedence: idx, benchmark_id: @benchmark&.id)
         end
 
-        ::Rule.import!(new_rules, ignore: true)
-      end
-
-      private
-
-      def split_rules
-        @split_rules ||= @rules.partition(&:new_record?)
-      end
-
-      def new_rules
-        @new_rules ||= split_rules.first
-      end
+        # This is needed for future computations with rule identifiers
+        @new_rules = @rules.select(&:new_record?)
```

Comment:
Hopefully this would not forget to recompute precedence when just new rules would be used.